### PR TITLE
Custom levelling patterns

### PR DESCRIPTION
**Pull request by mokonaDesu**
Implemented the custom levelling pattern for invoker. He gets all the skillpoints while other heroes start getting skillpoints once every 3 levels past level 25.

Also added -levelup chat command to levelup current hero quickly.

Fixes the other half of #282

### CODE REVIEW

File: game/scripts/vscripts/components/progression/hero_progression.lua [hunk 1]
```diff
@@ -1,48 +1,73 @@
 if HeroProgression == nil then
     HeroProgression = class({})
     Debug.EnabledModules['progression:*'] = true
+
```
Comment (mokonaDesu) [inline game/scripts/vscripts/components/progression/hero_progression.lua]: Fixed the indentation on the entire file - been using 4 spaces before, not its 2 spaces. That's why the changes are so many.

File: game/scripts/vscripts/components/progression/hero_progression.lua [hunk 2]
```diff
@@ -1,48 +1,73 @@
...
 
-function HeroProgression:Init ()
-    HeroProgression = self
+function HeroProgression:RegisterCustomLevellingPatterns()
+  HeroProgression.customLevellingPatterns = {}
+
+  HeroProgression.customLevellingPatterns['npc_dota_hero_invoker'] = (function(level)
+    -- Invoker gets all dem ability points
+    return true
+  end)
```
Comment (mokonaDesu) [inline game/scripts/vscripts/components/progression/hero_progression.lua]: Add overrides in this method if you need.

File: game/scripts/vscripts/components/progression/hero_progression.lua (hunk 1 above)
Comment (Chronophylos) [inline game/scripts/vscripts/components/progression/hero_progression.lua]: What editor are you using? It should support editorconfig.

Comment (mokonaDesu) [inline game/scripts/vscripts/components/progression/hero_progression.lua]: atom.io -- I've setup the editorconfig already, so it's gonna be 2 spaces from now on. I am using 4 spaces everywhere else, that's why.

Comment (Chronophylos) [inline game/scripts/vscripts/components/progression/hero_progression.lua]: Apparently 2 spaces is more modern..

Comment (chrisinajar) [inline game/scripts/vscripts/components/progression/hero_progression.lua]: It's just what's in our editorconfig. Add the .editorconfig extension to atom and it'll just automatically confirm to most of our basic settings
